pkg/apis/discovery/validation/validation.go

```diff
@@ -143,6 +143,12 @@ func validatePorts(endpointPorts []discovery.EndpointPort, fldPath *field.Path)
 		} else if !supportedPortProtocols.Has(string(*endpointPort.Protocol)) {
 			allErrs = append(allErrs, field.NotSupported(idxPath.Child("protocol"), *endpointPort.Protocol, supportedPortProtocols.List()))
 		}
+
+		if endpointPort.AppProtocol != nil {
+			for _, msg := range validation.IsValidLabelValue(*endpointPort.AppProtocol) {
```

Comment:
Confuisingly, you want `IsQualifiedName` - this one tests for values, you need keys.